L1Trigger/Phase2L1ParticleFlow/plugins/L1TCorrelatorLayer1Producer.cc

```diff
@@ -347,19 +357,29 @@ void L1TCorrelatorLayer1Producer::produce(edm::Event &iEvent, const edm::EventSe
     edm::Handle<std::vector<l1t::Vertex>> vtxHandle;
     iEvent.getByToken(extTkVtx_, vtxHandle);
     for (const auto &vtx : *vtxHandle) {
+      ptsums.push_back(std::pair<float,float>(vtx.pt(),vtx.z0()));
       if (ptsum == 0 || vtx.pt() > ptsum) {
         ptsum = vtx.pt();
         z0 = vtx.z0();
       }
     }
     pvwd = l1t::VertexWord(1, z0, 1, ptsum, 1, 1, 1);
   }
-
+  std::stable_sort(ptsums.begin(), ptsums.end(),[](const auto& a, const auto& b){return a.first > b.first;});
+  for(int i0 = 0; i0 < std::min(int(ptsums.size()),int(NVtx_)); i0++) { 
+    z0s.push_back(ptsums[i0].second);
+  } 
   l1ct::PVObjEmu hwpv;
   hwpv.hwZ0 = l1ct::Scales::makeZ0(pvwd.z0());
   event_.pvs.push_back(hwpv);
   event_.pvs_emu.push_back(pvwd.vertexWord());
-
+  //Do a quick histogram vertexing to get multiple vertices (Hack for the taus)
+  doVertexings(z0s);
+  for(unsigned int i = 1; i < z0s.size(); ++i) { 
+    l1ct::PVObjEmu hwpv; 
+    hwpv.hwZ0 = l1ct::Scales::makeZ0(z0s[i]);   
+    event_.pvs.push_back(hwpv); //Skip emu
+  }
```

Comment:
Can you put all this changes inside of an if nVtx > 1? 
So that we keep the existing functionality for nVtx = 1

Reply:
fixed